Microsoft.Azure.CosmosRepository/src/Builders/IItemBuilder.cs

```diff
@@ -11,18 +11,26 @@ namespace Microsoft.Azure.CosmosRepository.Builders
     /// <summary>
     /// A builder to configure a container for an item
     /// </summary>
-    public interface IItemContainerBuilder
+    public interface IItemBuilder
     {
         /// <summary>
         /// The list of already configured options for a <see cref="IItem"/>
         /// </summary>
-        IReadOnlyList<ContainerOptionsBuilder> Options { get; }
+        IReadOnlyList<ContainerOptionsBuilder> ContainerItemOptions { get; }
 
         /// <summary>
         /// Provides a <see cref="ContainerOptionsBuilder"/> instance to configure a container for an item
         /// </summary>
         /// <typeparam name="TItem">The type of <see cref="IItem"/> to configure.</typeparam>
-        /// <returns>Instance of <see cref="IItemContainerBuilder"/></returns>
-        IItemContainerBuilder Configure<TItem>(Action<ContainerOptionsBuilder> containerOptions) where TItem : IItem;
+        /// <returns>Instance of <see cref="IItemBuilder"/></returns>
+        IItemBuilder ConfigureContainer<TItem>(Action<ContainerOptionsBuilder> containerOptions) where TItem : IItem;
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="metadataBuilder"></param>
+        /// <typeparam name="TItem"></typeparam>
+        /// <returns></returns>
+        IItemBuilder ConfigureOptions<TItem>(Action<ItemOptionsBuilder> metadataBuilder) where TItem : IItem;
```

Comment:
I feel like the name `ConfigureOptions` is a bit too vague. Perhaps something to more clearly indicate that it lets you configure metadata, such as eTag, etc. Does `ConfigureItemPropertyMetadata` make more sense? I'm not really sure I like that name either. LOL

Reply:
Yes I started with with `ConfigureMetadata(...)` but then thought against it haha, what do you think?